### PR TITLE
Update telemetry events custom name

### PR DESCRIPTION
**Pull request by ousmaneo**
Update telemetry events

/nocl
/jpd https://github.com/Graylog2/graylog-plugin-enterprise/pull/5980
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/grok-patterns/BulkLoadPatternModal.jsx [hunk 1]
```diff
@@ -57,6 +60,12 @@ class BulkLoadPatternModal extends React.Component {
       GrokPatternsStore.bulkImport(request, importStrategy).then(() => {
         UserNotification.success('Grok Patterns imported successfully', 'Success!');
         this._closeModal();
+
+        this.props.sendTelemetry(TELEMETRY_EVENT_TYPE.GROK_PATTERN.IMPORTED, {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/grok-patterns/BulkLoadPatternModal.jsx:64]: Is this really a user event? I would imagine the user event is the user clicking something that will trigger the import. The import itself (successful or not) is not related to what the user does and should be logged somewhere else.

File: graylog2-web-interface/src/components/grok-patterns/EditPatternModal.jsx [hunk 2]
```diff
@@ -93,13 +97,18 @@ class EditPatternModal extends React.Component {
...
-    const { create } = this.props;
+    const { create, sendTelemetry } = this.props;
 
     this._closeModal();
 
     if (create) {
       this.setState({ name: '', pattern: '', sampleData: '', test_result: '' });
     }
+
+    sendTelemetry(TELEMETRY_EVENT_TYPE.GROK_PATTERN[create ? 'CREATED' : 'UPDATED'], {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/grok-patterns/EditPatternModal.jsx:108]: I think this should be tracked before anything else (setting state, closing modal) happens.

File: graylog2-web-interface/src/components/grok-patterns/BulkLoadPatternModal.jsx (hunk 1 above)
Comment (ousmaneo) [reply graylog2-web-interface/src/components/grok-patterns/BulkLoadPatternModal.jsx:64]: Yes, it is, in this new naming we send the user action on the entity 'Grok pattern imported' whenever possible, and when needed we also send the action (click etc.) like in the event definition.

File: graylog2-web-interface/src/components/sidecars/configuration-forms/CollectorForm.jsx [hunk 3]
```diff
@@ -16,9 +16,6 @@
  */
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configuration-forms/CollectorForm.jsx:16]: Can we migrate this to typescript?

File: graylog2-web-interface/src/components/sidecars/configuration-forms/CollectorForm.jsx [hunk 4]
```diff
@@ -43,32 +43,32 @@ ValidationMessage.propTypes = {
...
-  getInitialState() {
+  options.push({ value: 'linux', label: 'Linux' });
+  options.push({ value: 'windows', label: 'Windows' });
+  options.push({ value: 'darwin', label: 'Darwin' });
+  options.push({ value: 'freebsd', label: 'FreeBSD' });
+
+  return options;
+};
+
+class CollectorForm extends React.Component {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configuration-forms/CollectorForm.jsx:66]: And make it a functional component? :)

File: graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationForm.tsx [hunk 5]
```diff
@@ -95,12 +98,22 @@ const ConfigurationForm = ({
...
+    if (isCreate) {
+      promise = CollectorConfigurationsActions.createConfiguration(formData)
         .then(() => history.push(Routes.SYSTEM.SIDECARS.CONFIGURATION));
     } else {
-      CollectorConfigurationsActions.updateConfiguration(formData);
+      promise = CollectorConfigurationsActions.updateConfiguration(formData);
     }
+
+    promise.then(() => {
+      sendTelemetry(TELEMETRY_EVENT_TYPE.SIDECARS[`CONFIGURATION_${isCreate ? 'CREATED' : 'UPDATED'}`], {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationForm.tsx:112]: Should we track this only on resolved promise? The user clicked anyways.

File: graylog2-web-interface/src/components/sidecars/configurations/CollectorListContainer.jsx [hunk 6]
```diff
@@ -15,54 +15,66 @@
  * <http://www.mongodb.com/licensing/server-side-public-license>.
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configurations/CollectorListContainer.jsx:15]: Same here: Can we move it to ts? (And functional)?

File: graylog2-web-interface/src/components/sidecars/configurations/CollectorListContainer.jsx [hunk 7]
```diff
@@ -15,54 +15,66 @@
...
-  loadCollectors() {
-    CollectorsActions.list({});
-  },
+  handleClone = (collector, name, callback) => {
+    const { sendTelemetry } = this.props;
 
-  handleClone(collector, name, callback) {
     CollectorsActions.copy(collector, name)
       .then(() => {
+        sendTelemetry(TELEMETRY_EVENT_TYPE.SIDECARS.LOG_COLLECTOR_CLONED, {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configurations/CollectorListContainer.jsx:44]: Should we track this only on resolved promise? The user clicked anyways.

File: graylog2-web-interface/src/components/sidecars/configurations/CollectorListContainer.jsx [hunk 8]
```diff
@@ -15,54 +15,66 @@
...
+  };
 
-  handleDelete(collector) {
-    CollectorsActions.delete(collector);
-  },
+  handleDelete = (collector) => {
+    const { sendTelemetry } = this.props;
+
+    CollectorsActions.delete(collector).then(() => {
+      sendTelemetry(TELEMETRY_EVENT_TYPE.SIDECARS.LOG_COLLECTOR_DELETED, {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configurations/CollectorListContainer.jsx:57]: Should we track this only on resolved promise? The user clicked anyways.

File: graylog2-web-interface/src/components/sidecars/configurations/ConfigurationListContainer.jsx [hunk 9]
```diff
@@ -15,77 +15,98 @@
...
-  },
+  };
+
+  handleClone = (configuration, name, callback) => {
+    const { sendTelemetry } = this.props;
 
-  handleClone(configuration, name, callback) {
     CollectorConfigurationsActions.copyConfiguration(configuration, name)
       .then((response) => {
+        sendTelemetry(TELEMETRY_EVENT_TYPE.SIDECARS.CONFIGURATION_CLONED, {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configurations/ConfigurationListContainer.jsx:60]: Should we track this only on resolved promise? The user clicked anyways.

File: graylog2-web-interface/src/components/sidecars/configurations/ConfigurationRow.jsx [hunk 10]
```diff
@@ -33,18 +35,25 @@ class ConfigurationRow extends React.Component {
...
   };
 
   _handleDelete = () => {
     const { configuration, onDelete } = this.props;
 
     // eslint-disable-next-line no-alert
     if (window.confirm(`You are about to delete configuration "${configuration.name}". Are you sure?`)) {
-      onDelete(configuration);
+      onDelete(configuration).then(() => {
+        this.props.sendTelemetry(TELEMETRY_EVENT_TYPE.SIDECARS.CONFIGURATION_DELETED, {
```
Comment (grotlue) [inline graylog2-web-interface/src/components/sidecars/configurations/ConfigurationRow.jsx:52]: Should we track this only on resolved promise? The user clicked anyways.

File: graylog2-web-interface/src/pages/ViewEventDefinitionPage.tsx [hunk 11]
```diff
@@ -73,12 +74,11 @@ const ViewEventDefinitionPage = () => {
   }, [currentUser, history, params]);
 
   const handleDuplicateEvent = () => {
-    sendTelemetry('click', {
-      app_pathname: 'event-definition',
-      app_action_value: 'event-definition-action-copy',
-    });
-
     EventDefinitionsActions.copy(eventDefinition).then((duplicatedEvent) => {
+      sendTelemetry(TELEMETRY_EVENT_TYPE.EVENTDEFINITION_DUPLICATED, {
```
Comment (grotlue) [inline graylog2-web-interface/src/pages/ViewEventDefinitionPage.tsx:78]: Should we track this only on resolved promise? The user clicked anyways.

**Review by grotlue — CHANGES_REQUESTED** [review]
In general, there are quite some places where the tracking is fired when the promise is resolved. I think it makes more sense to track first thing in the event handler as we want to track user and not application behavior. I commented on some of them, but am not sure I got them all. Please have a look.

Besides that code-wise, it looks fine.

I created a list of all the places to test out of the review and will go on with testing now.

**Review by grotlue — CHANGES_REQUESTED** [review]
Code-wise it's fine! Thanks for the adjustments!

Below, find my testing results. I skipped a few tests because I either didn't know how to test it or thought it was a lot of setup. Please, let me know if you think I should have a look at those.
There are some tests where the tracking wasn't fired for me, could you have a look? The rest looks fine!

## Test result

**Legend**

✅ Tracking as expected | ❌ Tracking not as expected | ⏭️ Test skipped

---

✅ HTTPHeaderAuthConfig - Update Config
✅ Authentication - Select & Submit Authentication Backend
✅ Authentication - Group Sync - Click on Save button *
✅ Authentication - Server Config - Click on Next button in wizard *
✅ Authentication - User Sync - Click on Next button in wizard*
✅ System Cluster Overview - Change days in traffic overview graph
✅ URL Whitelist Form Modal - Submit form
⏭️ Configuration - Data Node - Certificate Renewal - Submit form
✅  Configuration - Data Node - Certificate Renewal Policy - Submit form
✅ Configuration - Index Sets Defaults - Submit form
✅  Configuration - Permissions - Submit form
✅ Configuration - Search - Submit form
✅ Configuration - URL Whitelist - Submit form
✅ Configuration - Users - Submit form
✅ Configuration - Decorators - Submit form
✅  Content Packs - Overview - Install
✅ Content Packs - Overview - Download
✅  Content Packs - Overview - Delete
✅ Alerts - Event Definition - Duplicate
✅  Alerts - Notifications - Submit creation form
✅ Alerts - Notifications - Submit edit form
✅ Alerts - Notifications - Select notification type
✅ Alerts - Notifications - Click execute test
✅ Alerts - Notifications - Overview - Bulk Actions - Delete
✅ Alerts - Notifications - Overview - Delete
✅ Alerts - Notifications - Overview - Test Notification
⏭️ GROK Patterns - Bulk load patterns modal - Patterns Imported
✅ GROK Patterns - Edit pattern modal - Submit creation form
✅ GROK Patterns - Edit pattern modal - Submit edit form
✅ GROK Patterns - Edit pattern modal - Test pattern
✅ Index Sets Creation Page - Create Index Set
✅ Index Sets Configuration Page - Save Configuration
✅ Index Sets - Overview - Set default
✅ Index Sets - Overview - Delete
✅ Inputs - Select Input Type
✅ Inputs - Create Input with selected type
✅ Inputs - Click on "Find more inputs" link
✅ Inputs - Overview - Delete
❌ Inputs - Overview - Edit -> No Tracking when clicking on "More Actions" > "Edit input"
✅ Inputs - Overview - Show received message
✅ Inputs - Overview - Manage extractors
✅ Inputs - Overview - Show metrics
✅ Inputs - Overview - Start
✅ Inputs - Overview - Stop
✅ Loggers - Select log level
✅ Loggers - Toggle show log metrics
✅ Lookup Tables - Caches - Submit creation form
✅ Lookup Tables - Caches - Submit edit form 
✅ Lookup Tables - Caches - Overview - Delete
✅ Lookup Tables - Data Adapter - Submit creation form
✅ Lookup Tables - Data Adapter - Submit edit form
✅ Lookup Tables - LUT - Delete
⏭️ Maps - Configuration - Update Geolocation config
✅ Outputs - Create
✅ Outputs - Assign
✅ Outputs - Remove globally
✅ Outputs - Remove from stream
✅ Outputs - Update
✅ Roles - Assign user
✅ Roles - Unassign user
⏭️ Sidecars - Collector Configuration Modal - Assign Sidecar config -> Couldn't find the Collector Configuration Modal
⏭️ Sidecars - Collector process action set -> Couldn't find where to do that
❌ Sidecars - Configuration - Collector - Select Process Management -> Tracking not fired, tested by visiting `/sidecars/configuration` clicking "Create Log Collector" and using the select with the according name
❌ Sidecars - Configuration - Collector - Select Operating System -> Tracking not fired, tested by visiting `/sidecars/configuration` clicking "Create Log Collector" and using the select with the according name
✅ Sidecars - Configuration - Submit creation form
✅ Sidecars - Configuration - Submit edit form
✅ Sidecars - Configuration - Collectors Overview - Clone
✅ Sidecars - Configuration - Collectors Overview - Delete
✅ Sidecars - Configuration - Overview - Clone
✅ Sidecars - Configuration - Overview - Delete
⏭️ Sidecars - Configuration - Delete -> Couldn't find another place to delete it than the overview
✅ ThreatIntel - Config - Save config
✅ Users - Create User - Submit form
✅ Users - Overview - Disable ser
✅ Users - Overview - Enable user
✅ Users - Overview - Delete

**Review by grotlue — APPROVED** [review]
Looking good! Tested the remaining cases and they all work now. Thanks a lot!

✅ Inputs - Overview - Edit 
✅ Sidecars - Configuration - Collector - Select Process Management
✅ Sidecars - Configuration - Collector - Select Operating System

**Pull request merged**